server/src/main/java/org/solarbank/server/service/CalculateService.java

```diff
@@ -17,26 +25,86 @@ public class CalculateService {
     public CalculateResult processCalculateRequest(
         PanelSize panelSize,
         Double panelEfficiency,
-        EnergyTariff energyTariff
+        EnergyTariff energyTariff,
+        Map<String, Double> nasaData
     ) {
+        Double panelArea = panelSize.getHeight() * panelSize.getWidth();
+
+        Table monthlyAverageTable = calculateEnergyGenPerMonth(nasaData, panelEfficiency, panelArea);
 
         Map<String, Double> energyGenPerMonth = new HashMap<>();
-        energyGenPerMonth.put("January", 0.1);
-        energyGenPerMonth.put("February", 0.2);
+        for (int i = 0; i < monthlyAverageTable.rowCount(); i++) {
+            String month = monthlyAverageTable.stringColumn("Month").get(i);
+            double energy = monthlyAverageTable.doubleColumn("Energy").get(i);
+            energyGenPerMonth.put(month, energy);
+        }
+
+        Double energyGenPerYear = monthlyAverageTable.doubleColumn("Energy").sum();
+        energyGenPerYear = Math.round(energyGenPerYear * 10.0) / 10.0;
+
+        Double savings = energyGenPerYear * energyTariff.getAmount();
 
         SavingsPerYear savingsPerYear = new SavingsPerYear();
         CurrencyUnit currencyUnit = Monetary.getCurrency(energyTariff.getCurrencyCode());
-        MonetaryAmount amount = Money.of(1000, currencyUnit);
+        MonetaryAmount amount = Money.of(savings, currencyUnit);
         savingsPerYear.setCurrencyCode(currencyUnit);
         savingsPerYear.setAmount(amount);
 
-        Double energyGenPerYear = 1.0;
-
         CalculateResult result = new CalculateResult();
         result.setEnergyGenPerMonth(energyGenPerMonth);
         result.setEnergyGenPerYear(energyGenPerYear);
         result.setSavingsPerYear(savingsPerYear);
 
         return result;
     }
-}
+
+    public Table calculateEnergyGenPerMonth(Map<String, Double> nasaData, Double panelEfficiency, Double panelArea) {
```

Comment:
Method for caluclating average monthly energy generation. Uses tablesaw, which either I'm not using efficiently or isn't very good